tests/integration/docusaurus/reference/core_concepts/checkpoints_and_actions.py

```diff
@@ -0,0 +1,434 @@
+validation_result_id_type = next(iter(validation_result_id_type_set))
+assert validation_result_id_type == ValidationResultIdentifier
+validation_result_id = results.run_results[[k for k in results.run_results.keys()][0]]
+assert (
+    type(validation_result_id["validation_result"]) == ExpectationSuiteValidationResult
+)
+assert type(results.checkpoint_config) == CheckpointConfig
+
+typed_results = {
+    "run_id": run_id_type,
+    "run_results": {
+        validation_result_id_type: {
+            "validation_result": type(validation_result_id["validation_result"]),
+            "actions_results": {
+                "<ACTION NAME FOR STORING VALIDATION RESULTS>": {
+                    "class": "StoreValidationResultAction"
+                }
+            },
+        }
+    },
+    "checkpoint_config": CheckpointConfig,
+    "success": True,
+}
+
+results = {
+    "run_id": RunIdentifier,
+    "run_results": {
+        ValidationResultIdentifier: {
+            "validation_result": ExpectationSuiteValidationResult,
+            "actions_results": {
+                "<ACTION NAME FOR STORING VALIDATION RESULTS>": {
+                    "class": "StoreValidationResultAction"
+                }
+            },
+        }
+    },
+    "checkpoint_config": CheckpointConfig,
+    "success": True,
+}
+
+assert typed_results == results
+
+# A few different Checkpoint examples
+os.environ["VAR"] = "ge"
+os.environ["MY_PARAM"] = "1000"
+os.environ["OLD_PARAM"] = "500"
+
+no_nesting = """
+name: my_checkpoint
+config_version: 1
+class_name: Checkpoint
+run_name_template: "%Y-%M-foo-bar-template-$VAR"
+validations:
+  - batch_request:
+      datasource_name: taxi_datasource
+      data_connector_name: default_inferred_data_connector_name
+      data_asset_name: yellow_tripdata_sample_2019-01
+      data_connector_query:
+        index: -1
+    expectation_suite_name: my_expectation_suite
+    action_list:
+      - name: store_validation_result
+        action:
+          class_name: StoreValidationResultAction
+      - name: store_evaluation_params
+        action:
+          class_name: StoreEvaluationParametersAction
+      - name: update_data_docs
+        action:
+          class_name: UpdateDataDocsAction
+    evaluation_parameters:
+      param1: $MY_PARAM
+      param2: 1 + $OLD_PARAM
+    runtime_configuration:
+      result_format:
+        result_format: BASIC
+        partial_unexpected_count: 20
+"""
+context.add_checkpoint(**yaml.load(no_nesting))
+results = context.run_checkpoint(checkpoint_name="my_checkpoint")
+assert results.success == True
+
+nesting_with_defaults = """
+name: my_checkpoint
+config_version: 1
+class_name: Checkpoint
+run_name_template: "%Y-%M-foo-bar-template-$VAR"
+validations:
+  - batch_request:
+      datasource_name: taxi_datasource
+      data_connector_name: default_inferred_data_connector_name
+      data_asset_name: yellow_tripdata_sample_2019-01
+      data_connector_query:
+        index: -1
```

Comment:
This index: -1 just looks broken to me when I read it. Is it required for the example to work? If we could remove this thing I'd consider it a win; open to some discussion on this point

Reply:
It's not required. I went back and forth on this as well. It is telling the data connector to use the most recent batch in the data asset, but that is the default behavior regardless so I will remove it.

Reply:
I have removed this option from all configs in this doc.